Travis test

jockebech opened the pull request.

jbech-linaro commented on hikey.mk:
```diff
@@ -84,28 +85,34 @@ arm-tf-clean:
 		cd $(EDK2_PATH); $(BASH) edksetup.sh; \
 		make -C $(EDK2_PATH)/BaseTools clean; \
-		make -C $(EDK2_PATH)/BaseTools; \
+		make -j1 -C $(EDK2_PATH)/BaseTools; \
```
`'-j1` is pointless, since that is the default it's better to use for example:
```
make -j `nproc` ...
```

That will give the number of cores on you device/computer you are using.

jbech-linaro commented on hikey.mk:
```diff
@@ -84,28 +85,34 @@ arm-tf-clean:
+		GCC49_AARCH64_PREFIX=$(AARCH64_CROSS_COMPILE) \
 		make -C $(EDK2_PATH) \
-			-f HisiPkg/HiKeyPkg/Makefile EDK2_ARCH=AARCH64 \
+			-j1 -f HisiPkg/HiKeyPkg/Makefile EDK2_ARCH=AARCH64 \
```
Same here.

jbech-linaro commented on hikey.mk:
```diff
@@ -118,9 +125,23 @@ linux-defconfig:
 		cd $(LINUX_PATH); \
+		LOCALVERSION= \
+		CROSS_COMPILE=$(AARCH64_CROSS_COMPILE) \
 		ARCH=arm64 scripts/kconfig/merge_config.sh \
```
I learnt something new here. I was aware of `scripts/config`, but not the `merge_config`, thanks!

jbech-linaro replied:
I learnt something new here. I was aware of scripts/config, but not the merge_config, thanks!

vchong replied:
Y/w! Yes, merge_config replaces 'make defconfig' in this case when there is an additional config file to be merged with defconfig.

jbech-linaro commented on hikey.mk:
```diff
@@ -132,30 +153,35 @@ linux-defconfig-clean:
+		CROSS_COMPILE=$(AARCH64_CROSS_COMPILE) \
+		ARCH=arm64 \
+		LOCALVERSION= \
+		-j`getconf _NPROCESSORS_ONLN` \
```
This does the same as I recommended above, however a more to type.

vchong replied:
Not sure why but parallel build generates an error, while removing the -j option builds fine.

jbech-linaro replied:
Hmm, that's a bit strange. But .. since this is about cleaning I would say it is perfectly OK to skip the -j parameter. I will hardly make any difference when cleaning.

vchong replied:
Sorry, I mean to write this for the uefi build, not linux.

jbech-linaro replied:
I see, well I don't think I've built  UEFI with -j flag before. UEFI is a bit scary in general :). I would imagine that they are missing some dependencies in their Makefile(s) that makes parallel builds fails. If we don't want to dig into it right now we could ask someone from LEG about it (Ard? Roy F? Leif?). I don't say we should ask them to find the problem. But more asking if this is a known problem.

vchong replied:
I know the uefi-build.sh script in git.linaro.org/uefi/uefi-tools.git works with parallel builds, but it calls directly build.py instead of make. I'll try asking LEG.

vchong replied:
So per Leif/LEG, this is a known issue but low priority for fix.

"The platform builds themselves already build parallel if you use uefi-build.sh.
Building BaseTools with -j  breaks horribly since the Makefiles are completely broken.
It's on the list of stuff we should fix at some point, but because the BaseTools only take
around 20s to build sequentially, and don't have to be rebuilt to rebuild the platform,
there are many more important things to work on."

jbech-linaro replied:
Right, so then just remove it.

vchong replied:
Yes, already removed in the rebase/squash before the merge, so we're ok.

jbech-linaro commented on hikey.mk:
```diff
@@ -296,51 +323,52 @@ filelist-tee: xtest
 
 busybox:
-	@if [ ! -d "$(GEN_ROOTFS_PATH)/build" ]; then \
+	@#if [ ! -d "$(GEN_ROOTFS_PATH)/build" ]; then
```
Is the comment intentional?

vchong replied:
The rebuild guard will cause an error after running busybox-cleaner from the previous commit, so I commented it out. busybox-cleaner from this commit should fix the issue, but I was wondering if I should just leave it commented out to be on the safe side, at the expense of longer build time. I think I'll just put it back in, i.e. remove the comment, for the next round of review.

jbech-linaro commented on hikey_debug.mk:
```diff
@@ -0,0 +1,384 @@
+BASH := $(shell which bash)
```
So, the big differences in this file are that we change some flags and point to other binaries? Can't we make those changes directly in `hikey.mk` and provide another make argument when we want to enable debug builds?
`make -f hikey.mk DEBUG=y ...`

vchong replied:
Since fvp.mk, on which hikey.mk is based on, only provided a release version, it seemed a bit cleaner to maintain a separate hikey_debug.mk than to clutter hikey.mk with a bunch of 'if..else', and also to keep hikey.mk as similar to fvp.mk as possible in terms of generic layout. If you prefer everything in a single makefile, I can do that too.

vchong replied:
Actually, after trying, the extra 'if..else' weren't that bad, i.e. not as many as expected.

jbech-linaro commented:
I think this is looking good now. Do you want to squash your patches and force push/update to a single patch?

vchong commented:
I'd built the images using this and wanted to run xtest first, but haven't had a chance to do it yet. Is it ok to squash/push/merge without testing first?

jbech-linaro commented:
> ... Not tested yet, still OK to merge?
It's OK on this particular git/branch, this isn't included in any official git yet (even though it might be the official branch soon).

vchong commented:
Yes, I didn't it's good to do that either, but I'll squash and force push just for this time since I won't be able to get to testing until I get back from pto. Think it should ok since like you said it's just cleaning up and not really a major change. If later testing reveals any issues, I'll send another PR for it.

vchong commented:
Addressed review comments.
Rebased and squashed.
Thanks!

vchong commented:
Tested ok.
24487 subtests of which 0 failed
43 test cases of which 0 failed
0 test case was skipped
TEE test application done!

jockebech committed:
gcc5: Update to use gcc5

- Adds legacy toolchain for use in older edk2 projects, this is the only
  project using the pre 5.x toolchain for now
- Removes the bare metal toolchain
- qemu: Change the path for libraries for GCC5
- fvp: compiler changes for ARM-TF and EDK2
       Update libteec path in initramfs
- mtk: Toolchain and libteec path updates
- juno: Toolchain and libteec path updates

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Joakim Bech <joakim.bech@linaro.org> (QEMU, FVP, HiKey)
Tested-by: Pascal Brand <pascal.brand@linaro.org> (QEMU)
Tested-by: Pascal Brand <pascal.brand@linaro.org> (FVP)

Commit:
Fix conditional include of driver module

Signed-off-by: Victor Chong <victor.chong@linaro.org>
Tested-by: Victor Chong <victor.chong@linaro.org> (QEMU with CONFIG_OPTEE=m)
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Joakim Bech <joakim.bech@linaro.org> (QEMU)

jockebech committed:
Use non-SSL version when downloading toolchains

In some cases we've seen errors in Travis saying that it fails to
download the toolchain due to a bad signature. To prevent that from
happening we're only downloading using standard http instead of https.

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>

jockebech committed:
hikey: Fixes compiler error for Debian builds

Related to: https://github.com/OP-TEE/optee_os/issues/832

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Joakim Bech <joakim.bech@linaro.org> (HiKey Debian)
Suggested-by: Machiry Aravind Kumar <makrvcs@gmail.com>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>

d3zd3z committed:
hikey: Use mtools to avoid sudo during build

Instead of using sudo/mount to build the FAT image, use commands from
the mtools package to copy files into the FAT image.  This eliminates
the requirement to use sudo during the build.  It also removes a small
sleep necessary to make umount work.

Signed-off-by: David Brown <david.brown@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>

jforissier committed:
hikey: recovery does not need sudo if udev is configured correctly

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: David Brown <david.brown@linaro.org>

Commit:
qemu_v8: add First build script for QEMU ARMv8-A

QEMU and OP-TEE on ARMv8 is working

Reviewed-by: Victor Chong <victor.chong@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Signed-off-by: Matt Ma <matt.ma@linaro.org>

Commit:
Add support for hello_world TA

Signed-off-by: Victor Chong <victor.chong@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: David Brown <david.brown@linaro.org>
Tested-by: Victor Chong <victor.chong@linaro.org> (qemu)
Tested-by: Joakim Bech <joakim.bech@linaro.org> (RPi3 with mods
to build_ta_helloworld_qemu.sh for a 64-bit build)

Commit:
hikey: fix bug: @echo: command not found

Changed some line formatting to make commands more robust as well.

Signed-off-by: Victor Chong <victor.chong@linaro.org>
Tested-by: Victor Chong <victor.chong@linaro.org> (hikey)
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>

seqlabs committed:
rpi3: Initial rpi3.mk file for Raspberry Pi 3

Besides the rpi3.mk file itself, this also contains the additional
kconfig files and flags used when building Linux and U-Boot.

A set of firmware images are also included.

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: David Brown <david.brown@linaro.org>

seqlabs committed:
rpi3: Add debugger config files

.gdbinit stores settings for loading the symbol files at the correct
addresses and pi3.cfg is the JTAG configuration file for OpenOCD.

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: David Brown <david.brown@linaro.org>

jenswikl committed:
qemu_v8: replace broken edk2 link

Replaces broken edk2 link with a slightly older but working link.

Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

jockebech committed:
mtk8173: Change path to libteec.so

When we started using GCC5.x the path for libraries was changed. The
same change is needed for MTK8173 also.

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>

jforissier committed:
qemu-check.exp: monitor TEE core log file and detect assertion failures

When "make check" is run and an assertion fails in TEE core, secure
world hangs, and therefore xtest hangs too. "make check" will not
report the failure and exit until a timeout occurs (480s).
In order to detect the error condition sooner, open the TEE core log
file and add a trigger on "assertion ... failed at ...".

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>

etienne-lms committed:
factorize 'filelist' and 'update_rootfs' rules

Factorize embedded file list and rootfs generation for fvp, juno,
mediatek, qemu, qemu_v8 and rpi3.

busybox is madatory to all 'common' rootfs.
optee-client, xtestn helloworld are always embedded: moved to common.

Remove obsolete CFG_DEV_PATH.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)
Tested-by: Joakim Bech <joakim.bech@linaro.org> (RPi3, MTK8173)
Tested-by: Etienne Carriere <etienne.carriere@linaro.org> (qemu_v7/v8)

etienne-lms committed:
cleanup op-tee core default log level

All targets already enable core debug log level: 3.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)
Tested-by: Joakim Bech <joakim.bech@linaro.org> (RPi3, MTK8173)
Tested-by: Etienne Carriere <etienne.carriere@linaro.org> (qemu_v7/v8)

etienne-lms committed:
qemu (ARMv7/v8): fix comment in qemu build script

Test 2001 is obsolete. xtest 1004 is a good candidate: load TA, invoke
and close.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)
Tested-by: Joakim Bech <joakim.bech@linaro.org> (RPi3, MTK8173)
Tested-by: Etienne Carriere <etienne.carriere@linaro.org> (qemu_v7/v8)

Commit:
Enable configuring QEMU with Virtfs support

Add support for configuring QEMU with VirtFS to share data
between the guest QEMU VM and host operating systems.

To specify dir, that has to be mounted inside QEMU VM, edit
QEMU_VIRTFS_HOST_DIR var in qemu.mk

To mount shared folder within QEMU ARM VM, run:
$ mount -t 9p -o trans=virtio host /tmp/shared/

If Virtfs support is enabled, libattr1-dev and libcap-dev
have to be installed before configuring QEMU.

For additional details see:
http://wiki.qemu.org/Documentation/9psetup

Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Tested-by: Igor Opaniuk <igor.opaniuk@linaro.org> (qemu_v7/v8)

Commit:
Fix toolchain.mk ROOT path

Change to relative path, the same as in common.mk

Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>

jforissier committed:
qemu: virtfs: fix wrong variable name

"make QEMU_VIRTFS_ENABLE=y" does not enable VirtFS because of a bad
variable name in common.mk.

Test case:

(host) echo Hello >/tmp/hello
(host) make -j8 run QEMU_VIRTFS_ENABLE=y QEMU_VIRTFS_HOST_DIR=/tmp
(QEMU) mkdir /tmp/host
(QEMU) mount_shared /tmp/host
(QEMU) cat /tmp/host/hello
Hello

Fixes: 97d0529c1687 ("Enable configuring QEMU with Virtfs support")
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

Commit:
toolchain.mk: Update legacy toolchain download url

Fixes https://github.com/OP-TEE/optee_os/issues/1195.

Signed-off-by: Victor Chong <victor.chong@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Joakim Bech <joakim.bech@linaro.org>

glneo committed:
dra7xx: Initial support for TI DRA7xx

Signed-off-by: Andrew F. Davis <afd@ti.com>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
[Rebase on top of master]
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

jforissier committed:
Compile head.bin from head.S

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

Commit:
Create uboot.env from uboot.txt

Add support for generation of uboot.env image from txt file

Fixes: https://github.com/OP-TEE/build/issues/85
Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>
Tested-by: Igor Opaniuk <igor.opaniuk@linaro.org> (RPi3)
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>

jockebech committed:
Fix incorrect UUID for hello world TA

With the recently fixed UUID changes (commit in optee_os:
(9a8a19cdba48dbca89c2a35d7984e9ec28f97bda) we missed to fix the hello
world TA.

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

Commit:
RPi3 parallel build fix

Fix parallel build issue (make -jn), now it works fine
Minor: add additional command for the proper cleanup of rpi3_firmware
target.

Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>

Commit:
U-Boot env bootargs cleanup

Add runtime kernel comandline creation (for both mmc/nfs boot)

Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>
Tested-by: Igor Opaniuk <igor.opaniuk@linaro.org>

Commit:
toolchain: update toolchain to support ARMv8.1

Update toolchain urls to 5.3-2016.05, which has support of ARMv8.1 arch
features, enabled by default on some setups with linux kernel v4.9 (rpi3)

ARMv8.1 architectural features in linux kernel:
CONFIG_ARM64_HW_AFDBM
CONFIG_ARM64_PAN
CONFIG_ARM64_LSE_ATOMICS
CONFIG_ARM64_VHE

Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>

jockebech committed:
hikey: Use common rules for TAs etc

It seems like there where some leftovers from the consolidation of the
filesystem generation patches a while ago. This cleans up duplicates
etc and fixes an error when generating the root fs due to recent
changes.

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>
Reviewed by: Victor Chong <victor.chong@linaro.org>

jockebech committed:
OP-TEE documentation refactoring

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Add rpi3.md

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Add hikey.md

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Add juno.md

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Add fvp.md

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Add qemu.md

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Add mtk8173.md

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Pull faq.md from Linaro website page

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Update links in faq.md to build.git instead of optee_os

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Moving all markdown files to a doc folder

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Pull debug.md from optee_os.git

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Update debug.md slightly

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

Commit:
rpi3: update firmware to newer version

Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jforissier committed:
docs: fix broken links to README.md

Use a relative path to README.md instead of a URL, so that the file for
the current HEAD is shown.

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>

Commit:
doc: Update RPi3 NFS/TFTP setup steps

Update RPi3 NFS/TFTP steps (current steps are out-dated and don't work)
Minor update of Upstream section

Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
hikey: Update HiKey Debian for 2.3.0 release

- Use a newer boot image (pre-built 16.06 UEFI image)
- Remove sudo requirement when using burn-boot
- Add udev rules help, similar to the hikey.mk file
- Step up OPTEE_PKG_VERSION to 2.3-0

Known issues:
- On HiKey 4G devices it seems like the partition table is to small to
  be able fit the newer 16.06 Debian root fs. It might be that we can
  tweak the partition table a bit, but for the moment we will keep using
  the 16.03 Debian root fs.

- When running this on the device
     $ dpkg --force-all -i /tmp/linux-image-*.deb
  there will be errors like this:
     blockdev: ioctl error on BLKROSET: Operation not permitted
  According to the comments in:
     https://github.com/OP-TEE/optee_os/issues/1234
  that message only shows up on the 16.03 root fs and it seems to only
  show up when running later kernel versions. However, if you reboot the
  device, everything is working, i.e, it will boot up your new kernel
  etc. We need to investige further to find the root cause for the error
  message and to implement a proper solution without error messages.

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>

Commit:
docs: rpi3: minor: fix url links

Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>

etienne-lms committed:
xtest: fix clean default configuration

Default clean configuration missed the output path information
to properly cleanup build artifacts.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

sylvainpelissier committed:
HiKey: remove targets aes-perf and sha-perf

aes-perf and sha-perf are replaced by xtest --aes-perf and
xtest --sha-perf, respectively.

Signed-off-by: Sylvain <sylvain.pelissier@gmail.com>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>
[Update commit message]
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>

Commit:
qemu: add option to enable SLiRP user networking

Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>
[Rebase on top of master]
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

jenswikl committed:
docs: Add automake to prerequisites

Acked-by: Jerome Forissier <jerome.forissier@linaro.org>
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

glneo committed:
Add dra7xx.md

Signed-off-by: Andrew F. Davis <afd@ti.com>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>

jenswikl committed:
docs: Add libssl-dev to prerequisites

libssl-dev is needed to compile the ARM-TF fiptool when compiling for Juno.

Acked-by: Jerome Forissier <jerome.forissier@linaro.org>
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

jforissier committed:
rootfs: start tee-supplicant on boot

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (QEMU)
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jenswikl committed:
qemu-check: test cases renamed to regression_*

Update expect script with new name of test the test cases.

Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

Commit:
hikey*.mk: Support ptable-linux-8g.img build

Fixes: https://github.com/OP-TEE/build/issues/64
Fixes: https://github.com/OP-TEE/optee_os/issues/1296
Signed-off-by: Victor Chong <victor.chong@linaro.org>
Tested-by: Victor Chong <victor.chong@linaro.org> (build only)
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
[Update commit message: turn "Fixes X" into "Fixes: X" tags]
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

Commit:
Fix for serverip issue

U-boot dhcp/bootp client changes "serverip" variable value in case if it
was specified in a dhcp/bootp server configuration.
Renamed serverip var, which now is used for nfsboot target to avoid
updating of this value by DHCP/BOOTP clients and just to use our
pre-defined static value.

Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Joakim Bech <joakim.bech@linaro.org> (RPi3)
Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>

jockebech committed:
Test

jockebech committed:
test

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>

jockebech committed:
Add apt sources

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>

jockebech committed:
Cleanup and prepare for FVP

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>

jockebech committed:
Testing ccache

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
